src/libs/ReportUtils.js

```diff
@@ -1426,28 +1426,39 @@ function buildOptimisticIOUReportAction(type, amount, currency, comment, partici
     };
 }
 
-function buildOptimisticReportPreview(reportID, iouReportID, payeeAccountID) {
-    return {
+function buildOptimisticReportPreview(chatReport, iouReport, existingReportPreviewAction, justPaidOutstandingIOU) {
```

Comment:
I think that passing `existingReportPreviewAction` goes against the single responsibility principle to just build the optimistic report preview here. I think we should just focus on building a preview here. We can do something like [this](https://github.com/Expensify/App/blob/c0d43662892ea8722cbb4918deba95e21664e49c/src/libs/actions/IOU.js#L339-L355) to get an existing one and update it if necessary

Reply:
Agree, made a similar comment [here](https://github.com/Expensify/App/pull/21578#discussion_r1246269743)

Reply:
yes, thank you both @Julesssss and @luacmartins! I cleaned it up as per your suggestions and it looks waaay better and easier to understand ❤️